### ISSUE TITLE
Marketing Cloud SDK not tracking user interactions on push notifications

### ISSUE THREAD

**Issue by byFadili** [allboatsrise/expo-marketingcloudsdk#17]
Hello !

I am using Marketing Cloud Mobile Push for push notification to send push notifications to registered devices and having some trouble tracking "Opened" and "Last activity date".

<img width="1275" alt="Screenshot 2023-07-11 at 15 45 15" src="https://github.com/allboatsrise/expo-marketingcloudsdk/assets/24880094/f5d886f9-36b7-4b32-b7fc-a84864d89f8d">

I tried the @allboatsrise/expo-marketingcloudsdk .track() method, but it doesn't seem to work.

On [MC SDK documentation](https://salesforce-marketingcloud.github.io/MarketingCloudSDK-iOS/analytics/analytics.html) they provide some information about tracking and I wanted to know if that has been implemented in this package.

Does anyone have the same issue ? 
Any idea why did is happening ?

Thank you !

**Comment by andrejpavlovic** [comment]
You can try adding this to `ExpoMarketingCloudSdkNotificationsDelegate.swift` to see if it helps.

```
  @objc
  public func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
    SFMCSdk.mp.setNotificationRequest(response.notification.request)
    completionHandler()
  }
```

**Comment by FelipeGCastro** [comment]
> You can try adding this to `ExpoMarketingCloudSdkNotificationsDelegate.swift` to see if it helps.
> 
> ```
>   @objc
>   public func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
>     SFMCSdk.mp.setNotificationRequest(response.notification.request)
>     completionHandler()
>   }
> ```

Hello @andrejpavlovic, tried to add the code but looks like it's not working 😞, another guess?

**Comment by andrejpavlovic** [comment]
@FelipeGCastro So you:
1. Receive the notification on your device
2. Press on the notification which then opens the app

After all this, the opened status is still not recorded?

**Comment by FelipeGCastro** [comment]
@andrejpavlovic , it took a long time to show on the overview, but looks like it works, also I forgot to set analyticsEnabled: true

**Comment by andrejpavlovic** [comment]
> also I forgot to set analyticsEnabled: true

That would definitely help :)

@FelipeGCastro Did you have to add the code above that I suggested as well?

**Comment by andrejpavlovic** [comment]
@FelipeGCastro I see you added it in https://github.com/allboatsrise/expo-marketingcloudsdk/pull/20 , works for me :)

**Comment by andrejpavlovic** [comment]
Fix published in [`49.0.1`](https://github.com/allboatsrise/expo-marketingcloudsdk/releases/tag/49.0.1
